### PR TITLE
Exhaus

### PR DESCRIPTION
**Pull request by jtlap**
adding exhaustive tests capabilities for one parameter float functors

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -48,6 +48,7 @@ NS_prevent_in_source_build()
 set(STF_STANDALONE_DESTINATION      ${PROJECT_SOURCE_DIR}/test)
 set(BRIGAND_STANDALONE_DESTINATION  ${PROJECT_SOURCE_DIR}/include/boost/simd/detail)
 set(NS_BENCH_STANDALONE_DESTINATION ${PROJECT_SOURCE_DIR}/bench)
+set(NS_EXHAUSTIVE_STANDALONE_DESTINATION ${PROJECT_SOURCE_DIR}/exhaustive)
```
Comment (jfalcou) [inline CMakeLists.txt]: this is related to the automatic update of external project, you don't need it here

File: cmake/ns/make_exhaustive.cmake [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                            EXCLUDE_FROM_ALL TRUE
+                          )
+
+    add_target_parent(${exhaustive})
+    add_dependencies(exhaustive ${exhaustive})
+  endforeach()
+endfunction()
+
+set(CMAKE_CXX_FLAGS_EXHAUSTIVE
+    " -O3 -fopenmp"
```
Comment (jfalcou) [inline cmake/ns/make_exhaustive.cmake]: this is g++ specific. You need to detect openMP using FinPackage and use the macro it returns to you. I'll give it a shot

File: exhaustive/exhaustive.hpp [hunk 3]
```diff
@@ -0,0 +1,310 @@
+// -------------------------------------------------------------------------------------------------
+/*!
+  @file Main header for exhaustive components
+  @copyright 2016 - NumScale SAS
+  Distributed under the Boost Software License, Version 1.0.
+  (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+**/
+// -------------------------------------------------------------------------------------------------
+
+#ifndef NS_BENCH_HPP_INCLUDED
```
Comment (jfalcou) [inline exhaustive/exhaustive.hpp]: wrong header

File: exhaustive/exhaustive.hpp [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+  s += const_t::value ? " const"  : "";
+  s += lref_t::value   ? "&"      : "";
+  s += rref_t::value   ? "&&"     : "";
+  return s;
+}
+template<typename T> inline std::string type_id( const T& )
+{
+  return type_id<T>();
+}
+}
```
Comment (jfalcou) [inline exhaustive/exhaustive.hpp]: indentation is all crappy

File: exhaustive/function/scalar/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+#   sqrt1pm1.cpp
+    )
+
+make_exhaustive("function.scalar" ${SOURCES})
+
+
+
+
+
+
```
Comment (jfalcou) [inline exhaustive/function/scalar/CMakeLists.txt]: Do you have stocks & bonds in a whitespace company ?

File: exhaustive/function/scalar/cbrt.cpp [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+//                 See accompanying file LICENSE.txt or copy at
+//                     http://www.boost.org/LICENSE_1_0.txt
+//==============================================================================
+#include <nt2/include/functions/cbrt.hpp>
+#include <nt2/include/constants/valmin.hpp>
+#include <nt2/include/constants/valmax.hpp>
+
+#include <nt2/sdk/exhaustive/exhaustive.hpp>
+
+#define NT2_ASSERTS_AS_TRAP
```
Comment (jfalcou) [inline exhaustive/function/scalar/cbrt.cpp]: this is useless and wrong

File: exhaustive/function/simd/log.cpp [hunk 7]
```diff
@@ -0,0 +1,38 @@
+//==============================================================================
+//         Copyright 2016        Numscale SAS
+//
+//          Distributed under the Boost Software License, Version 1.0.
+//                 See accompanying file LICENSE.txt or copy at
```
Comment (jfalcou) [inline exhaustive/function/simd/log.cpp:5]: please don't add those files if they're not ready please.

File: cmake/ns/make_exhaustive.cmake (hunk 2 above)
Comment (jtlap) [inline cmake/ns/make_exhaustive.cmake]: I don't know how to do that

Comment (jfalcou) [inline cmake/ns/make_exhaustive.cmake]: http://stackoverflow.com/questions/12399422/how-to-set-linker-flags-for-openmp-in-cmakes-try-compile-function

First answer

**Pull request merged**